src/test/java/com/olgaboiar/mint/ReaderTest.java

```diff
@@ -0,0 +1,56 @@
+package com.olgaboiar.mint;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ReaderTest {
+
+    @Test
+    void testReaderGetsInputHeadersForRequestWithHeadersOnly() throws IOException {
```

Comment:
Since the test class already implied what these tests are for, perhaps we can remove the duplicated "Reader" from the test itself?

Reply:
removed